File: scan_broken_imports.py
import os
import re

dirs = [
  'Service Admin/src',
  'Service Formateur/src',
  'Service Apprenant/src'
]

broken_files = []

# Pattern to find 'import {' followed by 'import {' on the next line (ignoring whitespace)
pattern = re.compile(r'import\s*{\s*[\r\n]+\s*import\s*{')

for base_dir in dirs:
    if not os.path.exists(base_dir):
        continue
    for root, _, files in os.walk(base_dir):
        for file in files:
            if file.endswith(('.tsx', '.ts')):
                full_path = os.path.join(root, file)
                try:
                    with open(full_path, 'r', encoding='utf-8') as f:
                        content = f.read()
                        if pattern.search(content):
                            broken_files.append(full_path)
                except:
                    pass

print("BROKEN_FILES:")
for f in broken_files:
    print(f)

File: scan_broken_imports.js
const fs = require('fs');
const path = require('path');

const dirs = [
  'Service Admin/src',
  'Service Formateur/src',
  'Service Apprenant/src'
];

const brokenFiles = [];

function scan(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      scan(fullPath);
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      const content = fs.readFileSync(fullPath, 'utf8');
      if (content.includes('import {\nimport {')) {
        brokenFiles.push(fullPath);
      }
    }
  }
}

dirs.forEach(d => {
    if (fs.existsSync(d)) scan(d);
});

console.log('BROKEN_FILES:');
brokenFiles.forEach(f => console.log(f));
